euphoria-flink/src/main/java/cz/seznam/euphoria/flink/batch/BatchFlowTranslator.java

```diff
@@ -140,19 +143,27 @@ protected FlowOptimizer createOptimizer() {
     // translate each operator to proper Flink transformation
     dag.traverse().map(Node::get).forEach(op -> {
       Operator<?, ?> originalOp = op.getOriginalOperator();
-      Translation<Operator<?, ?>> tx = translations.get(originalOp.getClass());
-      if (tx == null) {
+      List<Translation> txs = this.translations.get(originalOp.getClass());
+      if (txs.isEmpty()) {
         throw new UnsupportedOperationException(
                 "Operator " + op.getClass().getSimpleName() + " not supported");
       }
       // ~ verify the flowToDag translation
-      Preconditions.checkState(
-          tx.accept == null || Boolean.TRUE.equals(tx.accept.apply(originalOp)));
-
-      DataSet<?> out = tx.translator.translate(op, executorContext);
-
-      // save output of current operator to context
-      executorContext.setOutput(op, out);
+      Translation<Operator<?, ?>> firstMatch = null;
+      for (Translation tx : txs) {
+        if (tx.accept == null || Boolean.TRUE.equals(tx.accept.apply(originalOp))) {
```

Comment:
I know this has been here before, just want to ask if we know what is the added value of `Boolean.TRUE.equals` to simply `tx.accept.apply(...)`?

Reply:
Ah! Probably unboxing. Let's get rid of that optimization.